src/test/java/io/specto/hoverfly/ruletest/HoverflyDslWithStateTest.java

```diff
@@ -0,0 +1,82 @@
+package io.specto.hoverfly.ruletest;
+
+import io.specto.hoverfly.junit.rule.HoverflyRule;
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.HttpServerErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import static io.specto.hoverfly.junit.core.SimulationSource.dsl;
+import static io.specto.hoverfly.junit.dsl.HoverflyDsl.service;
+import static io.specto.hoverfly.junit.dsl.ResponseCreators.created;
+import static io.specto.hoverfly.junit.dsl.ResponseCreators.success;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+
+public class HoverflyDslWithStateTest {
+
+    @ClassRule
+    public static HoverflyRule hoverflyRule = HoverflyRule.inSimulationMode(dsl(
+        service("www.service-with-state.com")
+            .get("/api/bookings/create")
+            .willReturn(success().transitionsState("hasBeenAlreadyCreated", "true"))
```

Comment:
I don't know why we called this `transitionsState` in Hoverfly, but we don't need to make it 100% matching the JSON field name in the simulation file. At the end of the day, we want to make the JAVA DSL a bit easier to write and understand. 

So perhaps, the DSL with state should read like this:
```
service("www.service-with-state.com")
            .delete("/api/bookings/1")
            .withState("Booking", "Reserved")
            .withState("Voucher Redeemd", "true")
            .willReturn(success().andSetState("Booking", "Deleted").andRemoveState("Voucher Reddemed"))
```
`withState` instead of `requiresState`
`andSetState` instead of `transitionsState`
`andRemoveState` instead of `removeState` (because it is an action rather than as part of the response body) 

What do you think?

Reply:
Yes, this is much better.
I named it in that way since I wanted to keep the same format as it had been already used for the other parameters.